entity.go

```diff
@@ -391,22 +416,39 @@ func serializeStruct(src interface{}) ([]byte, error) {
 
 	se := getSerializationEncoder()
 	defer freeSerializationEncoder(se)
-	smd := &structMetaData{metaDatas: make([]string, 0, 16)}
-	_, fms := getFieldInfoAndMetadata(t) // Use this function to force generation if needed
-
-	if err := serializeStructInternal(se.enc, smd, fms, "", v); err != nil {
-		return nil, err
-	}
 
-	bufSize := se.buf.Len()
-	// final size = header + all metadatas + separators for metadata + data
-	finalBufSize := 1 + smd.totalLength + len(smd.metaDatas) + bufSize
-	finalBuf := make([]byte, finalBufSize)
-	finalBuf[0] = byte(serializationStateNormal)          // Set the header
-	serializeStructMetaData(finalBuf[1:], smd)            // Serialize the metadata
-	copy(finalBuf[finalBufSize-bufSize:], se.buf.Bytes()) // Copy the actual data
+	if ls, ok := src.(datastore.PropertyLoadSaver); ok {
+		se.buf.Write([]byte{serializationStatePropertyList})
```

Comment:
We should use the `WriteByte` method to skip the slice allocation. Also we should check for the error returned by this method.

Reply:
Fixed b5dc7dcd1dcd97975ca1b10366488c2d2a153059